File: programmers/N개의최소공배수.java
package programmers;

public class N개의최소공배수 {
	class Solution {
	    public int solution(int[] arr) {
	        int answer = lcm(arr[0], arr[1]);
	        
	        for(int i=2; i<arr.length; i++){
	            answer = lcm(answer, arr[i]);
	        }
	        
	        return answer;
	    }
	    
	    public int gcd(int a,int b) {
	        while(b!=0){
	            int r = a%b;
	            a=b;
	            b=r;
	        }
	        return a;
	    }
	    
	    public int lcm(int a, int b){
	        return a*b/gcd(a,b);
	    }
	    
	    
	}
}

File: JavaScript/programmers/N개의 최소공배수.js
function solution(arr) {
  let answer = arr[0];
  // 최대 공약수 구하기
  const gcd = (a, b) => {
    while (b > 0) {
      let temp = b;
      b = a % b;
      a = temp;
    }
    return a;
  };
  // 유클리드 호제법 두 수의 곱 / 최대공약수 = 최대 공배수
  const lcm = (a, b) => {
    return (a * b) / gcd(a, b);
  };

  for (let i = 1; i < arr.length; i++) {
    answer = lcm(answer, arr[i]);
  }
  return answer;
}
